index.html

```diff
@@ -4292,6 +4306,454 @@ <h3>Proof Formats</h3>
+Implementers might find these other specifications through other standards
+bodies or communities that utilize this specification.
+              </li>
+            </ol>
+          </li>
+          <li>
+Set <var>securingMechanismVerified</var>.<var>status</var> to the
+result of passing <var>inputBytes</var>, along with any relevant function-specific
+options, to the <var>verifyProof</var> function. For example, for
+Verifiable Credential Data Integrity [[?VC-DATA-INTEGRITY]] proof
+verification functions, the <var>expectedProofPurpose</var> option is set to
+`assertionMethod`, and if the <a>conforming document</a> is expected to be a
+<a>verifiable presentation</a>, the <var>domain</var> and <var>challenge</var>
+are also provided as options.
+          </li>
+          <li>
+If <var>securingMechanismVerified</var>.<var>status</var> is set to
+`false`, add a <a href="#CRYPTOGRAPHIC_SECURITY_ERROR">
+  CRYPTOGRAPHIC_SECURITY_ERROR</a> to
+<var>securingMechanismVerified</var>.<var>errors</var>.
+          </li>
+          <li>
+If <var>securingMechanismVerified</var>.<var>status</var> is set to
+`true`:
+            <ol class="algorithm">
+              <li>
+Set <var>securingMechanismVerified</var>.<var>document</var> by passing
+<var>inputBytes</var> to the <var>extractDocument</var> function.
+              </li>
+              <li>
+Set <var>securingMechanismVerified</var>.<var>mediaType</var> to the media type
+associated with <var>securingMechanismVerified</var>.<var>document</var>.
+              </li>
+            </ol>
+          </li>
+          <li>
+Return <var>securingMechanismVerified</var>.
+          </li>
+        </ol>
+
+      </section>
+
+      <section>
+        <h3>Problem Details</h3>
+
+        <p>
+When an implementation detects an anomaly while processing a document, a
+<dfn>ProblemDetails</dfn> object can be used to report the issue to other
+software systems. The interface for these types of objects follows [[RFC9457]]
+to encode the data. A <a>ProblemDetails</a> object consists of the following
+properties:
+        </p>
+
+        <dl>
+          <dt>type</dt>
+          <dd>
+The `type` <a>property</a> MUST be present and its value MUST be a <a>URL</a>
+identifying the type of problem.
+          </dd>
+          <dt>code</dt>
+          <dd>
+The `code` <a>property</a> is an OPTIONAL
+<a data-cite="RFC9457#name-extension-members">RFC9457 extension member</a> . If
+present, its value MUST be an integer that identifies the type of the problem.
+Integer codes are useful in systems that only provide integer return values.
+          </dd>
+          <dt>title</dt>
+          <dd>
+The `title` <a>property</a> MUST be present and its value SHOULD provide a short
+but specific human-readable string for the problem.
+          </dd>
+          <dt>detail</dt>
+          <dd>
+The `detail` <a>property</a> MUST be present and its value SHOULD provide a
+longer human-readable string for the problem.
+          </dd>
```

Comment:
How are these two properties internationalized?

Reply:
They're not, because they're JSON objects, and JSON doesn't have an internationalization story.

If we tried to enable internationalization here in the same way that we support it for VCs, we'd violate RFC 9457, since that RFC only allows strings for the `title` and `detail` properties.

Since this is information that's exposed to developers, it will be in whatever language the developer that wrote the software wants it to be in, which is typically English.

IOW, if we wanted to support internationalization here, we'd need to re-invent RFC 9457 and that would probably be frowned upon... where the benefit would be of dubious value.

Reply:
RFC 9457 allows for extensions as specified in https://www.rfc-editor.org/rfc/rfc9457.html#section-3.2

Could that mechanism be used for i18n? I'm imagining that this will come up during CR review if we don't address it here.

Reply:
> I'm imagining that this will come up during CR review

We've already been through CR review and have all the approvals to proceed. 

That said, this feature didn't exist then, so I'll pull in @aphillips and @r12a to provide any guidance on i18n requirements on return values for software libraries. Guys, what we have going on here is a question on whether or not return values from software libraries need to be able to support internationalization. The challenge here is that the IETF RFC that defines what we're using has no i18n support... and it's questionable if these human-readable strings are going to be exposed to anyone beyond developers.

Reply:
It looks like RFC 9457 uses HTTP's native localization to localize these fields. That is, ["the language used for human-readable strings (such as those in title and description) can be negotiated using the Accept-Language request header field"](https://www.rfc-editor.org/rfc/rfc9457.html#name-introduction) and I see `Content-Language` used in https://www.rfc-editor.org/rfc/rfc9457.html#name-the-problem-details-json-ob to describe the language of the problem details.

That's not directly useful to this spec, since you're not making requests to the securing specifications over HTTP, but it implies that a layer outside the problem details should handle language negotiation. Maybe it works to say that software implementing this should allow developers to configure a language?

Reply:
I assume its fine to return ProblemDetails with localization applied, if its not, then perhaps it would be better to just return a single code, and leave the richer reporting to another day.

Reply:
@OR13 wrote:
> I assume its fine to return ProblemDetails with localization applied

Yes, I would expect the same and for that to be driven by environmental variables, such as HTTP Headers like `Accept-Language`.